Enabled ASCII printing of Singularity Functions

arihantparsoya opened the pull request:
Fixes #11524. Added code for ASCII printing of Singularity Functions.

#### Sample Program
```python
>>> from sympy import SingularityFunction, pprint
>>> from sympy.abc import x, a, n
>>> pprint(SingularityFunction(x, a, n),use_unicode=False)
        n
<-a + x> 
```

* Tests are added.

arihantparsoya committed:
Enabled ASCII printing of Singularity Functions

Fixes #11524. Added code for ASCII printing of Singularity Functions.

Reference Function _print_pow().

Tests are added.

asmeurer commented on sympy/printing/pretty/pretty.py:
```diff
@@ -1151,7 +1151,10 @@ def _print_SingularityFunction(self, e):
         else:
-            return self._print_Function(e)
+            base = "<"
+            base += str(e.args[0]-e.args[1])
```
Use `self._print` to recursively print the expression.

parsoyaarihant replied:
using `base += self._print(e.args[0]-e.args[1])` gives `TypeError: Can't convert 'prettyForm' object to str implicitly`

Or did you mean the following 
```python
base += str(self._print(e.args[0]-e.args[1]))
````

parsoyaarihant replied:
Thanks, I got it.

arihantparsoya committed:
Used self._print() and self._print_seq()

Used `self._print_seq` to recursively print the Singularity function in
sequence.
